packages/app/src/cli/services/function/replay.ts

```diff
@@ -0,0 +1,27 @@
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {readFile} from '@shopify/cli-kit/node/fs'
+import {readdirSync} from 'fs'
+
+export async function readFunctionRunsDirectory(functionPath: string) {
+  // Determine folder to read for runs
+  // We'll need to update this to use the actual path when they are saved.
+  const runsFolder = joinPath(functionPath, 'runs')
+
+  // Read file names
+  // This might actually be a JSONL file
+  const runFileNames = readdirSync(runsFolder)
+
+  // full paths to read file
+  const runFilePaths = runFileNames.map((runFile) => joinPath(runsFolder, runFile))
+
+  // read contents
+  const runData = await Promise.all(
+    runFilePaths.map((runFile) => {
+      return readFile(runFile)
+    }),
+  )
+
+  // convert promise'd strings to json
+  const runs = runData.map((run) => JSON.parse(run))
```

Comment:
Instead of reading/parsing every file, since we're using individual files, we could probably get away with saving the files as `<uuid>.json` and returning an array of `{name: "<uuid>", updatedAt: Date}` using the file metadata.

Reply:
Hmmm interesting idea. We now have the `identifier` at the end of the file, so we should be able to produce this. I'll give it a whirl.